modules/aggregation-layers/src/aggregation-layer-v9/aggregation-layer.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  CompositeLayer,
+  LayerDataSource,
+  LayerContext,
+  UpdateParameters,
+  CompositeLayerProps,
+  Attribute,
+  AttributeManager
+} from '@deck.gl/core';
+import {Aggregator} from './aggregator';
+
+// TODO
+type GPUAggregator = Aggregator & {destroy(): void};
+// TODO
+type CPUAggregator = Aggregator;
+
+export type AggregationLayerProps<DataT> = CompositeLayerProps & {
+  data: LayerDataSource<DataT>;
+};
+
+export default abstract class AggregationLayer<
+  DataT,
+  ExtraPropsT extends {} = {}
+> extends CompositeLayer<Required<AggregationLayer<DataT>> & ExtraPropsT> {
+  static layerName = 'AggregationLayer';
+
+  state!: {
+    gpuAggregator: GPUAggregator | null;
```

Comment:
I wonder if rather than specifying CPU/GPU as options, the layer could instead accept a list of aggregators, and then pick whichever is compatible (similar to how luma does with the `Device`). We may want to add support for a `webGpuAggregator` in addition to `webGlAggregator`, or who knows what else in the future